'CompareTypes' is ambiguous

## Bug Description
<!--A clear and concise description of the problem (Note: A missing feature is not a bug).-->

There is a `CompareTypes` in MetaPhysicL and another `CompareTypes` in libMesh.

Compiler got confused sometimes

```
In file included from /Users/kongf/projects/coupling_xolotl/moose/framework/build/header_symlinks/MaterialPropertyInterface.h:16:
/Users/kongf/projects/coupling_xolotl/moose/framework/build/header_symlinks/MathUtils.h:124:10: error: reference to 'CompareTypes' is ambiguous
typename CompareTypes<T, T2>::supertype
     ^
/Users/kongf/projects/coupling_xolotl/moose/scripts/../libmesh/installed/include/libmesh/compare_types.h:115:8: note: candidate found by name lookup is 'libMesh::CompareTypes'
struct CompareTypes {
    ^
/Users/kongf/projects/coupling_xolotl/moose/scripts/../libmesh/installed/include/metaphysicl/dynamicsparsenumberbase.h:1193:20: note: candidate found by name lookup is 'std::CompareTypes'
using MetaPhysicL::CompareTypes;
```

## Steps to Reproduce
<!--Steps to reproduce the behavior (input file, or modifications to an existing input file, etc.)-->

Compile moose using this conda env

moose % conda list 
```
# packages in environment at /Users/kongf/miniconda3/envs/moose:
#
# Name                    Version                   Build  Channel
autoconf                  2.69            pl5320hbcb3906_10    conda-forge
automake                  1.16.2          pl5320h694c41f_3    conda-forge
beautifulsoup4            4.9.3              pyhb0f4dca_0    conda-forge
boost                     1.70.0           py37hbf1eeb5_1    conda-forge
boost-cpp                 1.70.0               h75728bb_2    conda-forge
bzip2                     1.0.8                hc929b4f_4    conda-forge
c-ares                    1.17.1               hc929b4f_0    conda-forge
ca-certificates           2020.12.5            h033912b_0    conda-forge
cctools                   949.0.1             hd9211c8_18    conda-forge
cctools_osx-64            949.0.1             h2f0f38f_18    conda-forge
certifi                   2020.12.5        py37hf985489_1    conda-forge
clang                     11.0.0               h694c41f_2    conda-forge
clang-11                  11.0.0          default_h9e6edd0_2    conda-forge
clang-tools               11.0.0          default_h9e6edd0_2    conda-forge
clang_osx-64              11.0.0               hb91bd55_5    conda-forge
clangxx                   11.0.0          default_h9e6edd0_2    conda-forge
clangxx_osx-64            11.0.0               h7e1b574_5    conda-forge
cloudpickle               1.6.0                      py_0    conda-forge
cmake                     3.19.1               h800fed6_0    conda-forge
compiler-rt               11.0.0               h01488ec_2    conda-forge
compiler-rt_osx-64        11.0.0               hd3c4e95_2    conda-forge
cppunit                   1.15.1               hb1e8313_0    conda-forge
curl                      7.71.1               hcb81553_8    conda-forge
cycler                    0.10.0                     py_2    conda-forge
cytoolz                   0.11.0           py37he6e4e01_1    conda-forge
dask-core                 2020.12.0          pyhd8ed1ab_0    conda-forge
dbus                      1.13.18              h18a8e69_0  
decorator                 4.4.2                      py_0    conda-forge
expat                     2.2.10               hb1e8313_2  
freetype                  2.10.4               h3f75d11_0    conda-forge
future                    0.18.2           py37hf985489_2    conda-forge
gettext                   0.19.8.1             h1f1d5ed_1    conda-forge
gfortran_impl_osx-64      7.3.0                hf4212f2_2    conda-forge
gfortran_osx-64           7.3.0                h22b1bf0_7    conda-forge
glib                      2.55.0                        0    conda-forge
gmp                       6.2.1                h2e338ed_0    conda-forge
hdf4                      4.2.13            h71d84a9_1004    conda-forge
hdf5                      1.10.5          nompi_h34ad4e8_1111    conda-forge
icu                       64.2                 h6de7cb9_1    conda-forge
imageio                   2.9.0                      py_0    conda-forge
isl                       0.19                          0    conda-forge
jinja2                    2.11.2             pyh9f0ad1d_0    conda-forge
jpeg                      9d                   hbcb3906_0    conda-forge
jsoncpp                   1.8.4             ha1b3eb9_1002    conda-forge
kiwisolver                1.3.1            py37h8ec247f_0    conda-forge
krb5                      1.17.2               h60d9502_0    conda-forge
latexcodec                2.0.1              pyh9f0ad1d_0    conda-forge
lcms2                     2.11                 h11f7e16_1    conda-forge
ld64                      530                 hd2e7500_18    conda-forge
ld64_osx-64               530                 hea264c1_18    conda-forge
ldid                      2.1.2                h7660a38_2    conda-forge
libblas                   3.9.0                3_openblas    conda-forge
libcblas                  3.9.0                3_openblas    conda-forge
libclang                  11.0.0          default_h9e6edd0_2    conda-forge
libclang-cpp11            11.0.0          default_h9e6edd0_2    conda-forge
libcurl                   7.71.1               h9bf37e3_8    conda-forge
libcxx                    11.0.0               h4c3b8ed_1    conda-forge
libedit                   3.1.20191231         h0678c8f_2    conda-forge
libev                     4.33                 haf1e3a3_1    conda-forge
libffi                    3.3                  h046ec9c_2    conda-forge
libgfortran               4.0.0               h50e675f_13    conda-forge
libgfortran4              7.5.0               h50e675f_13    conda-forge
libiconv                  1.15              h0b31af3_1006    conda-forge
liblapack                 3.9.0                3_openblas    conda-forge
libllvm11                 11.0.0               hf85e3d2_0    conda-forge
libllvm9                  9.0.1                ha1b3eb9_0    conda-forge
libnetcdf                 4.6.2             h1a02027_1003    conda-forge
libnghttp2                1.41.0               h7580e61_2    conda-forge
libopenblas               0.3.12          openmp_h63d9170_1    conda-forge
libpng                    1.6.37               h7cec526_2    conda-forge
libpq                     12.3                 h0cb8f33_3    conda-forge
libssh2                   1.9.0                h8a08a2b_5    conda-forge
libtiff                   4.1.0                h2ae36a8_6    conda-forge
libtool                   2.4.6             h2e338ed_1007    conda-forge
libuv                     1.40.0               h22f3db7_0    conda-forge
libwebp-base              1.1.0                hbcb3906_3    conda-forge
libxml2                   2.9.10               h53d96d6_0    conda-forge
libxslt                   1.1.33               h320ff13_0    conda-forge
livereload                2.6.3              pyh9f0ad1d_0    conda-forge
llvm-openmp               11.0.0               h73239a0_1    conda-forge
llvm-tools                11.0.0               hf85e3d2_0    conda-forge
lxml                      4.6.2            py37h52eb7ab_0    conda-forge
lz4-c                     1.8.3             h6de7cb9_1001    conda-forge
m4                        1.4.18            haf1e3a3_1001    conda-forge
mako                      1.1.3              pyh9f0ad1d_0    conda-forge
markupsafe                1.1.1            py37h395d20d_2    conda-forge
matplotlib                3.3.3            py37hf985489_0    conda-forge
matplotlib-base           3.3.3            py37hdacc966_0    conda-forge
mock                      4.0.3            py37hf985489_0    conda-forge
moose-mpich               3.3.2                   build_4    idaholab
moose-tools               2020.08.11       py37h125a3fb_0    idaholab
mpc                       1.1.0             ha57cd0f_1009    conda-forge
mpfr                      4.0.2                h72d8aaf_1    conda-forge
mpi                       1.0                       mpich    conda-forge
ncurses                   6.2                  h2e338ed_4    conda-forge
networkx                  2.5                        py_0    conda-forge
nspr                      4.20              h0a44026_1000    conda-forge
nss                       3.47                 hc0980d9_0    conda-forge
numpy                     1.19.4           py37hec87de9_2    conda-forge
olefile                   0.46               pyh9f0ad1d_1    conda-forge
openssl                   1.1.1j               hbcf498f_0    conda-forge
pandas                    1.1.5            py37h010c265_0    conda-forge
pcre                      8.44                 hb1e8313_0    conda-forge
perl                      5.32.0               hbcb3906_0    conda-forge
pillow                    8.0.1            py37h09f51b7_0    conda-forge
pip                       20.3.3             pyhd8ed1ab_0    conda-forge
pybtex                    0.23.0           py37hc8dfbb8_0    conda-forge
pylatexenc                2.7                pyh9f0ad1d_0    conda-forge
pyparsing                 2.4.7              pyh9f0ad1d_0    conda-forge
pyqt                      5.9.2            py37h2a560b1_4    conda-forge
python                    3.7.9           h6c3b2c9_0_cpython    conda-forge
python-dateutil           2.8.1                      py_0    conda-forge
python_abi                3.7                     1_cp37m    conda-forge
pytz                      2020.4             pyhd8ed1ab_0    conda-forge
pywavelets                1.1.1            py37h37391d0_3    conda-forge
pyyaml                    5.3.1            py37h395d20d_1    conda-forge
qt                        5.9.7                h8cf7e54_3    conda-forge
readline                  8.0                  h0678c8f_2    conda-forge
rhash                     1.3.6             haf1e3a3_1001    conda-forge
scikit-image              0.16.2           py37h4f17bb1_0    conda-forge
scipy                     1.5.3            py37h04d6967_0    conda-forge
setuptools                49.6.0           py37h2987424_2    conda-forge
sip                       4.19.8           py37h0a44026_0  
six                       1.15.0             pyh9f0ad1d_0    conda-forge
soupsieve                 2.0.1                      py_1    conda-forge
sqlite                    3.34.0               h17101e1_0    conda-forge
tapi                      1100.0.11            h9ce4665_0    conda-forge
tbb                       2020.2               h926bf3e_1    conda-forge
tk                        8.6.10               h0419947_1    conda-forge
toolz                     0.11.1                     py_0    conda-forge
tornado                   6.1              py37h4b544eb_0    conda-forge
vtk                       8.2.0           py37hcba847f_203    conda-forge
wheel                     0.36.2             pyhd3deb0d_0    conda-forge
xz                        5.2.5                haf1e3a3_1    conda-forge
yaml                      0.2.5                haf1e3a3_0    conda-forge
zlib                      1.2.11            h7795811_1010    conda-forge
zstd                      1.4.4                hed8d7c8_2    conda-forge
```

## Impact
Better 
<!--Does this prevent you from getting your work done, or is it more of an annoyance?-->

--- a/framework/include/utils/MathUtils.h
+++ b/framework/include/utils/MathUtils.h
@@ -121,7 +121,7 @@ template <
     typename std::enable_if<std::is_same<typename W<T>::index_type, unsigned int>::value &&
                                 std::is_same<typename W2<T2>::index_type, unsigned int>::value,
                             int>::type = 0>
-typename CompareTypes<T, T2>::supertype
+typename libMesh::CompareTypes<T, T2>::supertype
 dotProduct(const W<T> & a, const W2<T2> & b)
 {
   return a * b;
@@ -138,7 +138,7 @@ template <typename T,
                                       std::is_same<typename W2<T2>::index_type,
                                                    std::tuple<unsigned int, unsigned int>>::value,
                                   int>::type = 0>
-typename CompareTypes<T, T2>::supertype
+typename libMesh::CompareTypes<T, T2>::supertype
 dotProduct(const W<T> & a, const W2<T2> & b)
 {
   return a.contract(b);
@@ -159,7 +159,7 @@ dotProduct(const W<T> & a, const W2<T2> & b)
  */
 template <typename C,
           typename T,
-          typename R = typename CompareTypes<typename C::value_type, T>::supertype>
+          typename R = typename libMesh::CompareTypes<typename C::value_type, T>::supertype>
 R
 poly(const C & c, const T x, const bool derivative = false)
 {
@@ -194,7 +194,7 @@ poly(const C & c, const T x, const bool derivative = false)
  */
 template <typename C,
           typename T,
-          typename R = typename CompareTypes<typename C::value_type, T>::supertype>
+          typename R = typename libMesh::CompareTypes<typename C::value_type, T>::supertype>
 R
 polynomial(const C & c, const T x)
 {
@@ -215,7 +215,7 @@ polynomial(const C & c, const T x)
  */
 template <typename C,
           typename T,
-          typename R = typename CompareTypes<typename C::value_type, T>::supertype>
+          typename R = typename libMesh::CompareTypes<typename C::value_type, T>::supertype>
 R
 polynomialDerivative(const C & c, const T x)
 {

